File: src/1441-Build-an-Array-With-Stack-Operations/1441.go
func buildArray(target []int, n int) []string {
    res := []string{}
    k := 1
    
    for i := 0; i < len(target); i, k = i + 1, k + 1 {
        if target[i] == k {
            res = append(res, "Push")
        } else {
            i--
            res = append(res, "Push")
            res = append(res, "Pop")
        }
    }
    return res
}

File: src/1441-Build-an-Array-With-Stack-Operations/1441.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> res = new ArrayList();
        int k = 1;
        for (int i = 0; i < target.length; i++, k++) {
            if (target[i] == k) { 
                res.add("Push");
            } else { 
                i--;
                res.add("Push");
                res.add("Pop");
            }
        }
        return res;
    }
}